feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/data/repository/datasource/StakingRewardsSubqueryDataSourceImpl.kt

```diff
@@ -0,0 +1,45 @@
+package jp.co.soramitsu.feature_staking_impl.data.repository.datasource
+
+import jp.co.soramitsu.common.utils.networkType
+import jp.co.soramitsu.core.model.getSubqueryPath
+import jp.co.soramitsu.core_db.dao.StakingRewardDao
+import jp.co.soramitsu.core_db.model.TotalRewardLocal
+import jp.co.soramitsu.feature_staking_impl.data.mappers.mapStakingSubquerySumRewardResponseToAmount
+import jp.co.soramitsu.feature_staking_impl.data.mappers.mapStakingTotalRewardLocalToTotalReward
+import jp.co.soramitsu.feature_staking_impl.data.network.subscan.StakingApi
+import jp.co.soramitsu.feature_staking_impl.data.network.subscan.request.StakingSumRewardRequest
+import jp.co.soramitsu.feature_staking_impl.domain.model.TotalReward
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.withContext
+import java.math.BigInteger
+
+class StakingRewardsSubqueryDataSourceImpl(
```

Comment:
Suggestion: `StakingRewardsSubqueryDataSourceImpl` -> `SubqueryTakingRewardsDataSource`

We should try to avoid Hungarian notation (-Impl), since it just adds irrelevant noise to the naming. We only use it to name single implementation of interface for multi-module structure.
In this case we have `Subquery-` prefix, which indicates implementation details and thus, there is no need for `-Impl`

Reply:
Done